skiko/src/awtMain/kotlin/org/jetbrains/skiko/redrawer/Direct3DRedrawer.kt

```diff
@@ -23,9 +23,12 @@ internal class Direct3DRedrawer(
 
     init {
         val adapter = chooseAdapter(properties.adapterPriority.ordinal)
-        adapterName = getAdapterName(adapter)
+        if (adapter == 0L) {
+            throw RenderException("Failed to choose DirectX12 adapter.")
+        }
+        adapterName = getAdapterName(0)
         adapterMemorySize = getAdapterMemorySize(adapter)
-        onDeviceChosen(getAdapterName(adapter))
+        onDeviceChosen(adapterName)
```

Comment:
And adapterName = getAdapterName(0) will always take first adapter.
But adapterMemorySize = getAdapterMemorySize(adapter) takes from another one adapter.

I don't understand how it can fix bug?

Reply:
Yeah, I accidentally pushed the test code. it is fixed in the last commit.

`getAdapterName(0)` will crash, because 0 is not an index, it is pointer

Reply:
The fix in throwing exception. When we throw RenderException, fallback to OpenGL will happen